plugin.cpp

```diff
@@ -0,0 +1,319 @@
+/*
+ * FogLAMP "delta" filter plugin.
+ *
+ * Copyright (c) 2018 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Massimiliano Pinto
+ */
+
+#include <plugin_api.h>
+#include <config_category.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <strings.h>
+#include <string>
+#include <iostream>
+#include <filter_plugin.h>
+#include <filter.h>
+#include <reading_set.h>
+#include <map>
+#include <rapidjson/writer.h>
+
+#define FILTER_NAME "delta"
+#define DEFAULT_CONFIG "{\"plugin\" : { \"description\" : \"Delta filter plugin\", " \
+                       		"\"type\" : \"string\", " \
+				"\"default\" : \"" FILTER_NAME "\" }, " \
+			 "\"enable\": {\"description\": \"A switch that can be used to enable or disable execution of " \
+					 "the delta filter.\", " \
+				"\"type\": \"boolean\", " \
+				"\"default\": \"false\" }, " \
+			"\"config\" : {\"description\" : \"Delta filter configuration.\", " \
+				"\"type\" : \"JSON\", " \
+				"\"default\" : {}} }"
```

Comment:
You should really give an example here as the user has no idea what JSON to put here.